src/typescript.ts

```diff
@@ -27,29 +31,58 @@ function resolveTypescript(cwd: string): typeof import('typescript') {
   return ts
 }
 
-export function resolveTsConfig(cwd: string): null | TypescriptOptions {
-  let tsCompilerOptions: CompilerOptions = {}
-  let tsConfigPath: string | undefined
-  tsConfigPath = resolve(cwd, 'tsconfig.json')
-  if (fileExists(tsConfigPath)) {
-    const ts = resolveTypescript(cwd)
-    const basePath = tsConfigPath ? dirname(tsConfigPath) : cwd
-    const tsconfigJSON = ts.readConfigFile(tsConfigPath, ts.sys.readFile).config
-    tsCompilerOptions = ts.parseJsonConfigFileContent(
-      tsconfigJSON,
-      ts.sys,
-      basePath,
-    ).options
-  } else {
-    return null
-  }
-  return {
-    tsCompilerOptions,
-    tsConfigPath,
+function createResolveTsConfig() {
```

Comment:
We should be able to keep the previous function and create another helper like `cache(fn)`, and create a `cachedResolveTsConfig` function then use it everywhere. They don't have to be mixed